Simplified/Account.swift

```diff
@@ -149,7 +149,7 @@ private let accountSyncEnabledKey        = "NYPLAccountSyncEnabledKey"
     mainColor = authenticationDocument.colorScheme
     
     let registerUrl = authenticationDocument.links?.first(where: { $0.rel == "register" })?.href
-    if let url = registerUrl, url.hasPrefix("nypl.card-creator:") == true {
+    if let url = registerUrl, url.trimmingCharacters(in: .whitespacesAndNewlines).lowercased().hasPrefix("nypl.card-creator:") {
       supportsCardCreator = true
       cardCreatorUrl = String(url.dropFirst("nypl.card-creator:".count))
     } else {
```

Comment:
We should be able to save a lot of peripheral logic if we set `cardCreatorUrl` to `nil` on the condition that it does not form a proper URL. This would prevent the need for a URL check later down the line and cause `registrationIsPossible` to be false.

Reply:
Do you mean that if `registrationIsPossible` is false, the registration cell won't be there at all?

Reply:
so it seems like if we set `cardCreatorUrl` to nil, the code I changed in the VC won't be triggered. Which is good in a way (no crash). 

However is this ok functionality/product wise? The code looks like if the registerURL is not a card creator URL BUT is a valid URL, the app would still be able to register the user. It seems like the VC code was written with this use case in mind, so this would be a functionality change. Let me know if I'm wrong though, I don't know this code as well as you.

Reply:
In the way it's used now essentially, the `Account::cardCreatorUrl` really means `Account::signUpUrl`. It can be a "card creator URL" or a normal URL.

Reply:
Unit test `testAccountSetAuthenticationDocument` also suggests that the _"card creator not supported, but cardCreatorUrl not nil and valid"_ scenario is possible. Brooklyn library I think falls under this case, since it provides `https://go.bklynlibrary.org/ecard` as cardCreatorURL, which is a valid url, but that would set `supportsCardCreator = false`. So I pushed a new commit to make this explicit.